From m.kozlowski@tuxland.pl Mon Nov  6 15:33:01 2006
From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
To: Wolfgang =?iso-8859-2?q?M=FCes?= <wolfgang@iksw-muees.de>,
	linux-kernel@vger.kernel.org
Subject: USB: auerswald possible memleak fix
Date: Tue, 7 Nov 2006 00:31:51 +0100
Cc: Greg KH <gregkh@suse.de>
Message-Id: <200611070031.52051.m.kozlowski@tuxland.pl>

fix possible memory leak in auerbuf_setup().

Regards,

	Mariusz Kozlowski

Signed-off-by: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Signed-off-by: Wolfgang Muees <wolfgang@iksw-muees.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/auerswald.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/misc/auerswald.c
+++ gregkh-2.6/drivers/usb/misc/auerswald.c
@@ -780,7 +780,7 @@ static int auerbuf_setup (pauerbufctl_t 
 
 bl_fail:/* not enough memory. Free allocated elements */
         dbg ("auerbuf_setup: no more memory");
-	kfree(bep);
+	auerbuf_free(bep);
         auerbuf_free_buffers (bcp);
         return -ENOMEM;
 }
